server/common/nosql/mongo-dao/src/main/java/org/kaaproject/kaa/server/common/nosql/mongo/dao/model/MongoCredentials.java

```diff
@@ -0,0 +1,121 @@
+package org.kaaproject.kaa.server.common.nosql.mongo.dao.model;
+
+import org.kaaproject.kaa.common.dto.credentials.CredentialsDto;
+import org.kaaproject.kaa.common.dto.credentials.CredentialsStatus;
+import org.kaaproject.kaa.server.common.dao.model.Credentials;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.Version;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+import java.io.Serializable;
+import java.util.Arrays;
+
+import static org.kaaproject.kaa.server.common.nosql.mongo.dao.model.MongoModelConstants.*;
+
+@Document(collection = CREDENTIALS)
+public class MongoCredentials implements Credentials, Serializable {
+
+    private static final long serialVersionUID = 817998992561126368L;
+
+    @Id
+    private String id;
+    @Field(CREDENTIALS_BODY)
+    private byte[] credentialsBody;
+    @Field(CREDENTIAL_STATUS)
+    private CredentialsStatus status;
+    @Version
+    private Long version;
+
+    public MongoCredentials(CredentialsDto dto) {
+        this(dto, null);
+    }
+
+    public MongoCredentials(CredentialsDto dto, Long version) {
+        this.id = dto.getId();
+        this.credentialsBody = dto.getCredentialsBody();
+        this.status = dto.getStatus();
+        this.version = version;
+    }
+
+    public static long getSerialVersionUID() {
+        return serialVersionUID;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public byte[] getCredentialsBody() {
+        return credentialsBody;
+    }
+
+    public void setCredentialsBody(byte[] credentialsBody) {
+        this.credentialsBody = credentialsBody;
```

Comment:
You should copy arrays with [Arrays#copyOf(byte[], int)](https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#copyOf(byte[],%20int)) to prevent the field from changing by the backing array.